Linear_cell_complex/include/CGAL/draw_linear_cell_complex_function.h

```diff
@@ -0,0 +1,300 @@
+        {
+          if (!m_drawing_functor.volume_wireframe(*lcc, itv) &&
+              !m_drawing_functor.face_wireframe(*lcc, itv))
+          { compute_face(itv, it, lcc, m_drawing_functor, graphic_buffer); }
+          for(typename LCC::template Dart_of_cell_basic_range<2>::const_iterator
+                itf=lcc->template darts_of_cell_basic<2>(itv, markfaces).begin(),
+                itfend=lcc->template darts_of_cell_basic<2>(itv, markfaces).end();
+              itf!=itfend; ++itf)
+          {
+            lcc->mark(itf, markfaces);
+            if (!lcc->is_marked(itf, markedges) &&
+                m_drawing_functor.draw_edge(*lcc, itf))
+            {
+              compute_edge(itf, lcc, m_drawing_functor, graphic_buffer);
+              for(typename LCC::template Dart_of_cell_basic_range<1>::const_iterator
+                    ite=lcc->template darts_of_cell_basic<1>(itf, markedges).begin(),
+                    iteend=lcc->template darts_of_cell_basic<1>(itf, markedges).end();
+                  ite!=iteend; ++ite)
+              {
+                lcc->mark(ite, markedges);
+                if (!lcc->is_marked(ite, markvertices) &&
+                    m_drawing_functor.draw_vertex(*lcc, ite))
+                {
+                  compute_vertex(ite, lcc, m_drawing_functor, graphic_buffer);
+                  CGAL::mark_cell<LCC, 0>(*lcc, ite, markvertices);
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  for (typename LCC::Dart_range::const_iterator it = lcc->darts().begin(),
+         itend = lcc->darts().end(); it != itend; ++it)
+  {
+    lcc->unmark(it, markvertices);
+    lcc->unmark(it, markedges);
+    lcc->unmark(it, markfaces);
+    lcc->unmark(it, markvolumes);
+    lcc->unmark(it, oriented_mark);
+  }
+
+  lcc->free_mark(markvolumes);
+  lcc->free_mark(markfaces);
+  lcc->free_mark(markedges);
+  lcc->free_mark(markvertices);
+  lcc->free_mark(oriented_mark);
+}
+
+} // namespace draw_function
+
+/**
+ * @brief This function is responsible for filling the buffer to allow
+ * visualization.
+ *
+ * @param graphic_buffer
+ * @param m_drawing_functor
+ * @param alcc
+ */
+template <typename BufferType = float, class LCC, class DrawingFunctor>
+void add_in_graphic_buffer_lcc(const LCC &alcc, Graphic_buffer<BufferType> &graphic_buffer,
+                               const DrawingFunctor &m_drawing_functor)
+{
+  draw_function_for_lcc::compute_elements(&alcc, graphic_buffer, m_drawing_functor);
+}
+
+template <typename BufferType = float, class LCC>
+void add_in_graphic_buffer_lcc(const LCC &alcc, Graphic_buffer<BufferType> &graphic_buffer)
+{
+  Drawing_functor_with_volume<LCC,typename LCC::Dart_const_handle,
+                                  typename LCC::Dart_const_handle,
+                                  typename LCC::Dart_const_handle,
+                                  typename LCC::Dart_const_handle> drawing_functor_with_volume;
+
```

Comment:
here you can modify the drawing_functor_with_volume to change the default behavior. Maybe one random color per volume ? (have a look to the old code to see the previous implementation)

Reply:
Done.

Reply:
@gdamiand Ping.